File: blog-sever/server-java/src/main/java/com/site/blog/service/TagService.java
package com.site.blog.service;

import com.site.blog.model.entity.Tag;
import com.baomidou.mybatisplus.extension.service.IService;
import com.site.blog.model.entity.BlogTagCount;

import java.util.List;

/**
 * <p>
 * 标签表 服务类
 * </p>
 *
 */
public interface TagService extends IService<Tag> {

    /**
     * 让博客标签数指数
     *
     * @return {@link List}<{@link BlogTagCount}>
     */
    List<BlogTagCount> getBlogTagCountForIndex();

    /**
     * 清除标签
     *
     * @param tagId 标签id
     * @return boolean
     */
    int clearTag(String tagId);

}

File: myblog-kt/src/main/kotlin/com/site/blog/service/TagService.kt
package com.site.blog.service

import com.baomidou.mybatisplus.extension.service.IService
import com.site.blog.model.entity.BlogTagCount
import com.site.blog.model.entity.Tag

/**
 *
 *
 * 标签表 服务类
 *
 *
 */
interface TagService : IService<Tag?> {
    /**
     * 让博客标签数指数
     *
     * @return [List]<[BlogTagCount]>
     */
    fun getBogTagCountForIndex(): List<BlogTagCount>

    /**
     * 清除标签
     *
     * @param tagId 标签id
     * @return boolean
     */
    fun clearTag(tagId: String?): Int
}